### ISSUE TITLE
`jj log -p` and `jj log -p --git` do not wrap diffs correctly, even if `ui.log-word-wrap` is on

### ISSUE THREAD

**Issue by ilyagr** [martinvonz/jj#4160]
## Steps to Reproduce the Problem

1. `jj config set --user ui.log-word-wrap true`
1. `jj log -p` or `jj log -p --git` in a narrow terminal

## Actual Behavior

<img width="616" alt="image" src="https://github.com/user-attachments/assets/c518eff7-8957-42ab-88d2-1998e223e8ae">

## Expected

The output does not mess up the graph

**Comment by yuja** [comment]
iirc, it's intentional that word wrapping applies only to the template output. Wrapped unified diff is hard to read (and is kinda corrupted.)

Maybe we can add per-diff-style word wrapping options if needed?

**Comment by ilyagr** [comment]
I guess what you are saying is that if we do wrapping to fix the graph, then `less -S` will no longer work. Hmm, that's a problem.

Perhaps you are right, and this should be coupled with implementing word wrapping for regular `jj diff` and `jj diff --git`.

**Comment by yuja** [comment]
I meant wrapped color-words diff would look ugly for example:
```
 123 123: very loooooong
line
```
ideally, it should be rendered as
```
 123 123: very loooooong
        : line
```

And unified diff is a machine-readable format. User might expect parsable output in `log --no-graph`.

**Comment by ilyagr** [comment]
> I meant wrapped color-words diff would look ugly for example:

This doesn't look too bad to me. I think `difftastic` does something like it.

> User might expect parsable output in log --no-graph.

I didn't mean this issue to affect `--no-graph`.

I'm still worried about making it harder to use `less -S`, though.

**Comment by yuja** [comment]
> This doesn't look too bad to me. I think `difftastic` does something like it.

No. `difft` by default shows contents side by side. It would be even uglier if `difft --width=100` output were wrapped at 80 columns.

> > User might expect parsable output in log --no-graph.
> 
> I didn't mean this issue to affect `--no-graph`.

Perhaps, it'll be a separate config knob then. `ui.log-word-wrap=true` should apply to both graph/no-graph outputs. If graph diff output were wrapped, no-graph output should do for consistency.

**Comment by ilyagr** [comment]
> No. difft by default shows contents side by side. It would be even uglier if difft --width=100 output were wrapped at 80 columns.

You are correct, only the side-by-side mode of difft wraps AFAICT.

> Perhaps, it'll be a separate config knob then. ui.log-word-wrap=true should apply to both graph/no-graph outputs. If graph diff output were wrapped, no-graph output should do for consistency.

If we'd like that, we could disable wrapping for `--no-graph --git` if the output is not a terminal.

This also raises another point: it seems that before addressing wrapping diffs for `jj log -p`, we'd want to implement wrapping diffs for `jj diff` and/or `jj diff --git` in a way that respects all these issues. That seems possible (we could add nice line-wrapping indicators even), but not extremely valuable.

So, on one hand, I think all these issues could be addressed. On the other hand, you made it clear that this would not be trivial to address, and the current behavior has the advantage of working with `less -S`, so I'm not planning to rush into it anytime soon.
